pkg/kube/inject/inject.go

```diff
@@ -302,9 +302,17 @@ func injectRequired(ignored []string, config *Config, podSpec *corev1.PodSpec, m
 // ProxyImage constructs image url in a backwards compatible way.
 // values based name => {{ .Values.global.hub }}/{{ .Values.global.proxy.image }}:{{ .Values.global.tag }}
 func ProxyImage(values *opconfig.Values, image *proxyConfig.ProxyImage, annotations map[string]string) string {
-	imageName := "proxyv2"
 	global := values.GetGlobal()
 
+	imageName := "proxyv2"
+	if global.GetProxy() != nil && global.GetProxy().GetImage() != "" {
+		imageName = global.GetProxy().GetImage()
+		if strings.Contains(imageName, "/") {
+			// here we handle both proxy.image and proxy_init.image. This can happen when proxy_init.image is not set, but proxy.image is
+			return imageName
+		}
+	}
+
```

Comment:
```suggestion
	imageName := global.GetProxy().GetImage()
	if global.GetProxy() != nil && imageName != "" {
		if strings.Contains(imageName, "/") {
			// here we handle both proxy.image and proxy_init.image. This can happen when proxy_init.image is not set, but proxy.image is
			return imageName
		} 
		return "proxyv2"
	}
```

Reply:
This wouldn't work:
- If we return `proxyv2` we don't build the '<hub>/<name>:<tag>`, which is the expected output of the function in the case where `proxy.image` isn't a URL